packages/venia-concept/src/components/FilterModal/FilterFooter/filterFooter.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import catalogActions, { serialize } from 'src/actions/catalog';
+import { withRouter } from 'react-router-dom';
+import classify from 'src/classify';
+import defaultClasses from './filterFooter.css';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+class FilterFooter extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            resetButton: PropTypes.string,
+            resetButtonDisabled: PropTypes.string,
+            applyButton: PropTypes.string,
+            applyButtonDisabled: PropTypes.string,
+            footer: PropTypes.string
+        }),
+        history: PropTypes.object,
+        filterClear: PropTypes.func,
+        chosenFilterOptions: PropTypes.object,
+        closeModalHandler: PropTypes.func
+    };
+
+    static getDerivedStateFromProps(nextProps) {
+        const { chosenFilterOptions } = nextProps;
+        let filterOptionsArePristine = true;
+
+        for (const optionName in chosenFilterOptions) {
+            if (chosenFilterOptions[optionName].length > 0) {
+                filterOptionsArePristine = false;
+                break;
+            }
+        }
+
+        return { areOptionsPristine: filterOptionsArePristine };
+    }
+
+    state = {
+        areOptionsPristine: true
+    };
+
+    resetFilterOptions = () => {
+        const { history, filterClear } = this.props;
+        history.push();
+        filterClear();
+    };
+
+    handleApplyFilters = () => {
+        const { history, chosenFilterOptions, closeModalHandler } = this.props;
+        history.push('?' + serialize(chosenFilterOptions));
+        closeModalHandler();
+    };
+
+    getFooterButtons = areOptionsPristine => {
+        const { classes } = this.props;
+
+        const resetButtonClass = areOptionsPristine
```

Comment:
Instead of using `getDerivedStateFromProps` we can simply check if there are any `chosenFilterOptions` when we render since the footer component is "connected" to the catalog store in redux.

Reply:
This code remained from previous contributor. I have resolved it in latest update.